app/assets/javascripts/discourse/tests/unit/services/toasts-test.js

```diff
@@ -0,0 +1,21 @@
+import { getOwner } from "@ember/application";
+import { setupTest } from "ember-qunit";
+import { module, test } from "qunit";
+
+module("Unit | Service | Toasts", function (hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.toasts = getOwner(this).lookup("service:toasts");
+  });
+
+  test("views option", async function (assert) {
+    this.toasts.show({ views: ["desktop"], data: { text: "foo" } });
+
+    assert.deepEqual(this.toasts.activeToasts.length, 1);
+
+    this.toasts.show({ views: ["mobile"], data: { text: "foo" } });
+
+    assert.deepEqual(this.toasts.activeToasts.length, 1);
```

Comment:
Maybe I am not reading this right but this test doesn't seem correct -- unless site.mobileView is true a toast shouldn't be added to `activeToasts`? Or in this case is it just because the first `toast.show` call still hasn't been dismissed and been removed from `activeToasts`, so we are actually checking that `this.toasts.activeToasts.length` is **not 2**?

Reply:
yes that's it we are checking that it didn't reach 2.

I will change it to: assert.ok(this.toasts.activeToasts.length < 2);

Reply:
Thanks for that :heart: